code/modules/mining/ores_coins.dm

```diff
@@ -466,23 +466,70 @@ GLOBAL_LIST_INIT(sand_recipes, list(\
 	to_chat(user, span_notice("You detach the string from the coin."))
 	return TRUE
 
-/obj/item/coin/attack_self(mob/user)
+/obj/item/coin/proc/flip(mob/user, flash = FALSE)
 	if(cooldown < world.time)
 		if(string_attached) //does the coin have a wire attached
-			to_chat(user, span_warning("The coin won't flip very well with something attached!") )
+			if(user)
+				to_chat(user, span_warning("The coin won't flip very well with something attached!") )
 			return FALSE//do not flip the coin
 		coinflip = pick(sideslist)
 		cooldown = world.time + 15
+
 		flick("coin_[cmineral]_flip", src)
 		icon_state = "coin_[cmineral]_[coinflip]"
-		playsound(user.loc, 'sound/items/coinflip.ogg', 50, 1)
+		if(flash)
+			SSvis_overlays.add_vis_overlay(src, icon, "flash", ABOVE_LIGHTING_LAYER, ABOVE_LIGHTING_PLANE, unique = TRUE)
+		playsound(loc, 'sound/items/coinflip.ogg', 50, TRUE)
 		var/oldloc = loc
 		sleep(1.5 SECONDS)
+		SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
 		if(loc == oldloc && user && !user.incapacitated())
 			user.visible_message("[user] has flipped [src]. It lands on [coinflip].", \
  							 span_notice("You flip [src]. It lands on [coinflip]."), \
 							 span_italics("You hear the clattering of loose change."))
 	return TRUE//did the coin flip? useful for suicide_act
 
+/obj/item/coin/attack_self(mob/user)
+	flip(user)
+
+/obj/item/coin/after_throw(datum/callback/callback) //get rid of the throw rotation
+	. = ..()
+	transform = initial(transform)
+
+/obj/item/coin/bullet_act(obj/item/projectile/P)
+	if(P.flag == "laser" || P.flag == "energy" || istype(P, /obj/item/projectile/bullet/c38)) //energy projectiles get deflected (also det revolver because damn thats cool)
+		if(cooldown >= world.time)//we ricochet the projectile
+			var/list/targets = list()
+			var/turf/center = get_turf(src)
+			for(var/mob/living/T in viewers(world.view - 3, src))
```

Comment:
1. What does ``T`` mean? Can you use a better var name?
2. Why the ``world.view - 3``? It would be better to use a hardcoded radius so you can control how far people will notice it from. Yeah it might be on your screen, but if world.view is so high, how would you notice it from so far?

Reply:
1.T stands for target, and it's simply a way to distinguish temporary variables, i didn't make this convention
2. I guess i'll make it static